File: code/bit_manipulation/src/Subarray with given XOR/subarray_with_given_xor.java
  public class Solution {
    public int solve(ArrayList<Integer> A, int k) {
       int count=0;
       int xr=0; // current xor
       HashMap<Integer,Integer> map=new HashMap<>(); //stores y wrt count
       //y=xr^k
       for(int a:A) {
            xr^=a;
            if(map.containsKey(xr^k)) //y=xr ^ k
                count+=map.get(xr^k);
            if(xr==k) count++;

            map.put(xr,map.getOrDefault(xr,0)+1);

       }
       return count;
    }
}

File: code/bit_manipulation/src/Subarray with given XOR/subarray_with_given_xor.cpp
// Part of Cosmos by OpenGenus Foundation
int Solution::solve(vector<int> &nums, int target) {
   
    // we have to check how many subarrays having xor of target
   
    // if any subarray has xor value of total , if total ^ target = p , & p is present in the map
   
    // then this will give one answer
   
    unordered_map< int  , int > mp;
   
   // mp[0]=1;
   
    int xor_val=0;
   
    int count=0;
   
    for(int i=0;i<nums.size();i++)
    {
        xor_val = xor_val ^ nums[i];
       
        if(xor_val==target)
        {
            count++;
        }
       
        if(mp.find(xor_val ^ target)!=mp.end())
        {
            count+=(mp[xor_val ^ target]);
        }
       
        mp[xor_val]++;
    }
   
    return count;
   
}
